src/libraries/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceFactory.byte.cs

```diff
@@ -37,7 +41,11 @@ internal class MemoryTestSequenceFactory : ReadOnlySequenceFactory<T>
         {
             public override ReadOnlySequence<T> CreateOfSize(int size)
             {
-                return CreateWithContent(new T[size]);
+#if DEBUG
+                return new ReadOnlySequence<T>(new ReadOnlyMemory<T>(new T[size + 1]).Slice(1));
+#else
+                return new ReadOnlySequence<T>(new ReadOnlyMemory<T>(new T[size]));
+#endif
```

Comment:
I generally prefer for the ifdef to be around the minimal thing that's different, so that it's clear what exactly the difference is.  e.g.
```C#
ReadOnlyMemory<T> m = new T[size + 1];
#if DEBUG
m = m.Slice(1);
#endif
return new ReadOnlySequence<T>(m);
```